cms/search/test_wagtail_hooks.py

```diff
@@ -0,0 +1,51 @@
+from unittest.mock import patch
+
+from django.test import TestCase
+
+from cms.search.wagtail_hooks import EXCLUDED_PAGE_TYPES, page_published, page_unpublished
+
+
+class MockRequest:
+    """Simulates the request passed to the wagtail hook."""
+
+    pass
+
+
+class MockPage:
+    """A minimal Page-like mock."""
+
+    def __init__(self, class_name, url_path="/some/path"):
+        self.url_path = url_path
+        self.__class__.__name__ = class_name
+
+
+class WagtailHooksTests(TestCase):
+    @patch("cms.search.wagtail_hooks.publisher")
+    def test_page_published_excluded_page_type(self, mock_publisher):
+        """Pages in EXCLUDED_PAGE_TYPES should not trigger any publish calls."""
+        for excluded_type in EXCLUDED_PAGE_TYPES:
+            page = MockPage(excluded_type)
+            page_published(MockRequest(), page)
```

Comment:
please use [`get_dummy_request()`](https://github.com/wagtail/wagtail/blob/bfcd0c7496279137c28ebd00a304fa2a514df47b/wagtail/coreutils.py#L413)

Reply:
Done + using the factories we have now.